src/client/app/components/unit/EditUnitModalComponent.tsx

```diff
@@ -132,17 +193,33 @@ export default function EditUnitModalComponent(props: EditUnitModalComponentProp
 		}
 	};
 
-	/* Edit Unit Validation:
-		Name cannot be blank
-		Sec in Rate must be greater than zero
-		Unit type mismatches checked on submit
-		If type of unit is suffix their must be a suffix
-	*/
-	const [validUnit, setValidUnit] = useState(false);
+	// Stores if save should be allowed but check for use by a meter is delayed until
+	// save is hit to avoid doing a lot and to give error message then.
+	const [canSave, setCanSave] = useState(false);
+	// Keeps canSave state up to date. Checks if valid and if edit made.
 	useEffect(() => {
-		setValidUnit(state.name !== '' && state.secInRate > 0 &&
-			(state.typeOfUnit !== UnitType.suffix || state.suffix !== ''));
-	}, [state.name, state.secInRate, state.typeOfUnit, state.suffix]);
+		// This checks:
+		// - Name cannot be blank
+		// - If type of unit is suffix their must be a suffix
+		// - The rate is set so not the custom input value. This happens if select custom value but don't input with enter.
+		// - The custom rate is a positive integer
+		const validUnit = state.name !== '' &&
+			(state.typeOfUnit !== UnitType.suffix || state.suffix !== '') && state.secInRate !== Number(CUSTOM_INPUT)
```

Comment:
After looking at comments on the CreateUnitModalComponent, I noticed that the design of the current implementation was to discard all values when leaving the page without saving. Given this, the custom rate would be discarded alongside the other values despite changing the similar line of code in CreateModalComponent. Should CreateUnitModalComponent be changed to still keep unsaved changes similarly to how it is implemented in EditUnitModalComponent?

Reply:
I thought this test was to see if the user had started a custom input but not hit enter. This likely indicates they started to enter a value but never finalized it with an enter. This seems a potentially common mistake that OED is trying to catch to avoid. I didn't think it has to do with leaving the page and resetting state but maybe I'm missing something. What do you think?

Reply:
Hi Steve, I changed the code to check for enter. The only caveat is that I made it purposely allow users to save without clicking enter if they didn't change the custom value from the standard values automatically inputted when clicked (ex. day to custom value, box shows 86400, create unit can be clicked). Let me know if this doesn't seem ideal and I'll change it to require enter to create a unit under all circumstances.

Reply:
Sorry I missed this comment until now. I'm not sure which is best. It is unclear why the admin selected custom if they did not intend to change the value. However, I'm okay with this as long as both create and edit work the same.

Reply:
I think the difference won't be noticeable for the user. Should I change it to require a change of value or is it fine as is?

Reply:
I'm not sure if I understand but it is okay to use the value without enter if it is the standard value as long as both create and edit do the same thing. Is this what you were asking?

Reply:
Yes. Do you have a preference between the two implementations?

Reply:
I don't care too much but I might pick that if they do custom then they have to hit enter. One case I am thinking about is if they change the value away from a standard one and then put it back without hitting enter then would the code accept this second value because it is standard? This type of issue is why I'm leaning toward requiring enter for all custom values but maybe it isn't an issue.

Reply:
Okay, I've edited it so it requires both that the custom value isn't a standard rate and that enter is pressed. I believe this makes it line up with edit.